src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.FullGlobalizationData.cs

```diff
@@ -66,7 +66,33 @@ private static unsafe bool TryConvertWindowsIdToIanaId(string windowsId, string?
             }
 
             char* buffer = stackalloc char[100];
-            int length = Interop.Globalization.WindowsIdToIanaId(windowsId, region, buffer, 100);
+
+            // regionPtr will point at the region name encoded as ASCII.
+            IntPtr regionPtr = IntPtr.Zero;
+
+             // Regions usually are 2 or 3 characters length.
+            const int MaxRegionNameLength = 11;
+
+            // Ensure uppercasing the region as ICU require the region names be uppercased, otherwise ICU will assume default region and return unexpected result.
+            if (region is not null && region.Length < MaxRegionNameLength)
+            {
+                byte* regionInAscii = stackalloc byte[region.Length + 1];
+                int i = 0;
+                for (; i < region.Length && region[i] <= '\u007F'; i++)
```

Comment:
Seems like this loop condition assumes the name won't have mixed case... ?

Reply:
`region[i] <= '\u007F'` is covering the whole ASCII range. So, this will include all upper and lower cased alphabet letters.  having `Au` is going to be successfully converted to `AU`. I'll add a test case for that.

Reply:
Yup, you're right.  I was reading it wrong.